CONTRIBUTING.md

```diff
@@ -116,6 +117,52 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 ```
 You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `--signoff` to add the `Signed-off-by` line to the end of the commit message.
 
+## Changelog
+
+OpenSearch maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+
+
+### How to add my changes to CHANGELOG](CHANGELOG.md)?
+
+As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
+
+Adding in the change is two step process -
+1. Add your changes to the corresponding section within the CHANGELOG file with dummy pull request information, publish the PR
+
+  `Your change here ([#PR_NUMBER](PR_URL))`
+
+2. Update your changes with the corresponding `PR_NUMBER` and  `PR_URL` and push out the new commit.
+
+For example, as a contributor, my change adds support for a new `FooBar` client, the changes would look like -
+
+```
+# CHANGELOG
+Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+
+## [ UNRELEASED ]
+### Added
+- ...
+- Add support for FooBar client ([#9999](https://github.com/opensearch-project/OpenSearch/pull/9999))
```

Comment:
Is it possible to change the format to include the contributor alias? I want to give people easy credit for the work they do.

Reply:
as i already wrote in https://github.com/opensearch-project/security/pull/1821#discussion_r865756545 i personally don't think listing the contributors is a good idea. when reading the changelog i want to know what changed and what it means for me - any other information is just annoying clutter and will be ignored and slows me down. i can always follow the links or check the commit history to know more.

Reply:
We could potentially use github aliases within the format, but like the PRs, it would be a self update process. 
+1 on following the PR links / browsing commit history to reduce information overload.